pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -122,331 +110,293 @@ def license_is_valid():
 
     def _apply_solver(self):
         if not self._save_results:
-            for block in self._pyomo_model.block_data_objects(descend_into=True,
-                                                              active=True):
-                for var in block.component_data_objects(ctype=pyomo.core.base.var.Var,
-                                                        descend_into=False,
-                                                        active=True,
-                                                        sort=False):
+            for block in self._pyomo_model.block_data_objects(
+                descend_into=True,active=True):
+                for var in block.component_data_objects(
+                    ctype=pyomo.core.base.var.Var, descend_into=False,
+                    active=True,sort=False):
                     var.stale = True
+
         if self._tee:
             def _process_stream(msg):
                 sys.stdout.write(msg)
                 sys.stdout.flush()
             self._solver_model.set_Stream(
-                self._mosek.streamtype.log, _process_stream)
+                self._mosek.streamtype.log,_process_stream)
 
         if self._keepfiles:
-            print("Solver log file: "+self._log_file)
+            print("Solver log file : " + self._log_file)
 
         for key, option in self.options.items():
-
-            param = self._mosek
-
             try:
-                for sub_key in key.split('.'):
-                    param = getattr(param, sub_key)
-            except (TypeError, AttributeError):
-                raise
-
-            if 'sparam' in key.split('.'):
-                self._solver_model.putstrparam(param, option)
-            else:
-                if 'iparam' in key.split('.'):
-                    self._solver_model.putintparam(param, option)
+                param = ".".join(("self._mosek",key))
+                if 'sparam' in key.split('.'):
+                    self._solver_model.putstrparam(eval(param),option)
                 elif 'dparam' in key.split('.'):
-                    self._solver_model.putdouparam(param, option)
-                else:
-                    raise AttributeError(
-                        "Unknown parameter type. Type sparam, iparam or dparam expected.")
-
-        self._termcode = self._solver_model.optimize()
-        self._solver_model.solutionsummary(self._mosek.streamtype.msg)
-
-        # FIXME: can we get a return code indicating if Mosek had a significant failure?
-        return Bunch(rc=None, log=None)
+                    self._solver_model.putdouparam(eval(param),option)
+                elif 'iparam' in key.split('.'):
+                    if isinstance(option, str):
+                        if 'mosek' in option.split('.'):
```

Comment:
I do this because integer parameters accept enumerations, such as `mosek.onoffkey.on`  (see [here](https://docs.mosek.com/9.2/pythonapi/parameters.html#integer-parameters)). So my rationale was that if a user reads about the accepted values to a parameter from MOSEK's online documentation, and decides to use them verbatim, they should be able to.
I don't check for it in the other types of parameters because they accept either doubles or strings.